keyboards/minimon/bartlesplit/matrix.c

```diff
@@ -0,0 +1,152 @@
+static const pin_t col_pins[] = MATRIX_COL_PINS;
+
+static void select_row(uint8_t row) {
+    setPinOutput(row_pins[row]);
+    writePinLow(row_pins[row]);
+}
+
+static void unselect_row(uint8_t row) {
+    setPinInputHigh(row_pins[row]);
+}
+
+static void unselect_rows(void) {
+    for (uint8_t x = 0; x < MATRIX_ROWS; x++) {
+        setPinInputHigh(row_pins[x]);
+    }
+}
+
+static void select_col(uint8_t col) {
+    setPinOutput(col_pins[col]);
+    writePinLow(col_pins[col]);
+}
+
+static void unselect_col(uint8_t col) {
+    setPinInputHigh(col_pins[col]);
+}
+
+static void unselect_cols(void) {
+    for (uint8_t x = 0; x < MATRIX_COLS/2; x++) {
+        setPinInputHigh(col_pins[x*2]);
+    }
+}
+
+static void init_pins(void) {
+    unselect_cols();
+    unselect_rows();
+}
+
+static bool read_cols_on_row(matrix_row_t current_matrix[], uint8_t current_row) {
+    // Store last value of row prior to reading
+    matrix_row_t last_row_value = current_matrix[current_row];
+
+    // Select row and wait for row selection to stabilize
+    select_row(current_row);
+    wait_us(30);
+
+    // For each col...
+    for (uint8_t col_index = 0; col_index < MATRIX_COLS / 2; col_index++) {
+        uint16_t column_index_bitmask = ROW_SHIFTER << ((col_index * 2) + 1);
+        // Check row pin state
+        if (readPin(col_pins[col_index*2])) {
+            // Pin HI, clear col bit
+            current_matrix[current_row] &= ~column_index_bitmask;
+        } else {
+            // Pin LO, set col bit
+            current_matrix[current_row] |= column_index_bitmask;
+        }
+    }
+
+    // Unselect row
+    unselect_row(current_row);
+
+    return (last_row_value != current_matrix[current_row]);
+}
+
+static bool read_rows_on_col(matrix_row_t current_matrix[], uint8_t current_col) {
+    bool matrix_changed = false;
+
+    // Select col and wait for col selection to stabilize
+    select_col(current_col*2);
+    wait_us(30);
+
+    // For each row...
+    for (uint8_t row_index = 0; row_index < MATRIX_ROWS; row_index++) {
+        // Store last value of row prior to reading
+        matrix_row_t last_row_value = current_matrix[row_index];
+
+        uint16_t column_index_bitmask = ROW_SHIFTER << (current_col * 2);
+        // Check row pin state
+        if (readPin(row_pins[row_index])) {
+            // Pin HI, clear col bit
+            current_matrix[row_index] &= ~column_index_bitmask;
+        } else {
+            // Pin LO, set col bit
+            current_matrix[row_index] |= column_index_bitmask;
+        }
+
+        // Determine if the matrix changed state
+        if ((last_row_value != current_matrix[row_index]) && !(matrix_changed)) {
+                   matrix_changed = true;
+        }
+    }
+
+    // Unselect col
+    unselect_col(current_col*2);
+
+    return matrix_changed;
+}
+
+
+void matrix_init_custom(void) {
+    // initialize key pins
+    init_pins();
+
+    debounce_init(MATRIX_ROWS);
```

Comment:
```suggestion
```

Reply:
I'm guessing we can also just remove `#include "debounce.h"` then?